analyzers/src/SonarAnalyzer.CSharp/Rules/BlazorQueryParameterTypeShouldBeSupported.cs

```diff
@@ -0,0 +1,106 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+namespace SonarAnalyzer.Rules.CSharp;
+
+[DiagnosticAnalyzer(LanguageNames.CSharp)]
+public sealed class BlazorQueryParameterTypeShouldBeSupported : SonarDiagnosticAnalyzer
+{
+    private const string DiagnosticId = "S6797";
+    private const string MessageFormat = "Query parameter type '{0}' is not supported.";
+
+    private static readonly DiagnosticDescriptor Rule = DescriptorFactory.Create(DiagnosticId, MessageFormat);
+
+    private static readonly ISet<KnownType> SupportedTypes = new HashSet<KnownType>
+    {
+        KnownType.System_Boolean,
+        KnownType.System_DateTime,
+        KnownType.System_Decimal,
+        KnownType.System_Double,
+        KnownType.System_Single,
+        KnownType.System_Int32,
+        KnownType.System_Int64,
+        KnownType.System_String,
+        KnownType.System_Guid
+    };
+
+    public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+    protected override void Initialize(SonarAnalysisContext context) =>
+        context.RegisterCompilationStartAction(cc =>
+        {
+            if (cc.Compilation.GetTypeByMetadataName(KnownType.Microsoft_AspNetCore_Components_RouteAttribute) is null)
+            {
+                return;
+            }
+
+            cc.RegisterSymbolAction(c =>
+            {
+                var componentClass = (INamedTypeSymbol)c.Symbol;
+                if (HasRouteAttribute(componentClass))
+                {
+                    foreach (var propertyType in GetPropertyTypeMismatches(componentClass))
+                    {
+                        c.ReportIssue(Diagnostic.Create(Rule, propertyType.GetLocation(), GetTypeName(propertyType)));
+                    }
+                }
+            },
+            SymbolKind.NamedType);
+        });
+
+    private static string GetTypeName(TypeSyntax propertyType)
+    {
+        if (propertyType.NameIs(KnownType.System_Nullable_T.TypeName))
+        {
+            propertyType = ((GenericNameSyntax)propertyType).TypeArgumentList.Arguments[0];
```

Comment:
Educational: are we sure at this point that there is at least 1 argument?

Reply:
I checked the type symbol before in `IsSupportedType` so it should be fine.
But I will add more test with a custom define type named `Nullable`

Reply:
That was a nice catch!
With a custom type named Nullable, it fails!